plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitManager.java

```diff
@@ -137,11 +138,16 @@ private List<BigQuerySplit> readFromBigQuery(ConnectorSession session, TableDefi
             // Storage API doesn't support reading materialized views and external tables
             return ImmutableList.of(BigQuerySplit.forViewStream(columns, filter));
         }
-        if (isSkipViewMaterialization(session) && type == VIEW) {
-            return ImmutableList.of(BigQuerySplit.forViewStream(columns, filter));
+        if (type == VIEW) {
+            if (isSkipViewMaterialization(session)) {
+                return ImmutableList.of(BigQuerySplit.forViewStream(columns, filter));
+            }
+            tableConstraint.getDomains().ifPresent(domains -> domains.keySet().stream()
+                    .map(column -> ((BigQueryColumnHandle) column).getName())
+                    .forEach(projectedColumnsNames::add));
```

Comment:
Otherwise, `TestBigQueryAvroConnectorTest.testEmptyProjectionView` throws the following error:
```
io.grpc.StatusRuntimeException: INVALID_ARGUMENT: request failed: Query error: Unrecognized name: regionkey at [1:105]
```